runner/catalog_api.go

```diff
@@ -0,0 +1,11 @@
+package runner
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func CatalogHandler(runnerService RunnerService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.JSON(200, runnerService.GetCatalog())
```

Comment:
This returns always `200` state, even though you request the catalog and it is not ready.
We can modify it of course